src/typechecker.erl

```diff
@@ -1170,17 +1162,17 @@ infer_literal_string(Str, Env) ->
                                {char, erl_anno:new(0), lists:last(SortedChars)}])])
     end.
 
-expect_tuple_type({type, _, any, []}, _N) ->
+expect_tuple_type({type, _, any, []}, _N, _Env) ->
```

Comment:
The `_Env` parameter that you've added doesn't seem to be used anywhere. 
I suggest adding a spec to this function, while you're at it.

Reply:
It's used in only one clause, but this clause is the reason for introducing `Env`:

```
expect_tuple_type(?user_type() = Ty, N, Env) ->
    expect_tuple_type(normalize(Ty, Env), N, Env);
```

Reply:
It's one of those places where we need to call `normalize()` explicitly now.

Reply:
Spec added :+1: